### ISSUE TITLE
[BUG] Upgrade (v1.1.2 -> v1.2.0-rc6) stuck in pre-drained

### ISSUE THREAD

**Issue by lanfon72** [harvester/harvester#4477]
### Describe the bug
![image](https://github.com/harvester/harvester/assets/5169694/e5b1a1a1-8c10-40bf-ac85-a72164e5ab47)
![image](https://github.com/harvester/harvester/assets/5169694/9b241c0a-c1a3-4813-bfbe-b554526e4f87)
![image](https://github.com/harvester/harvester/assets/5169694/b5659d20-abd4-4215-8161-3f5877ab6582)
![image](https://github.com/harvester/harvester/assets/5169694/0daa6e73-3aba-4beb-9c64-cec304c1c85a)

### To Reproduce
1. Install Harvester with at least 4 nodes
2. Create an image for VM creation
3. Create VLAN
4. Create 2 VMs with 1 snapshot and 1 backup
5. Import Harvester into Rancher 2.7.6
6. Create RKE1 and RKE2 cluster (1 node)
7. Install Harvester Cloud Provider and CSI Driver
8. Create DHCP load balancer service (only RKE1 cluster)
9. Create Nginx deployment with new Harvester PVC
10. Upgrade Harvester

### Expected behavior
Cluster should be upgraded successfully

## Environment:
 - Harvester ISO version: **v1.1.2** -> **v1.2.0-rc6**
 - **ui-source** Option: **Auto**
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): **Baremetal DL360 4 nodes**

### Additional context
![image](https://github.com/harvester/harvester/assets/5169694/19d91927-1504-4e76-84fe-7d2f51ad2b15)
![image](https://github.com/harvester/harvester/assets/5169694/6b42ec91-8213-48a6-917c-777dd9564694)

### Support bundle
> 25MB, unable to upload

**Comment by lanfon72** [comment]
## Support Bundle
Nodes:
[harvester-lhlkv.zip](https://github.com/harvester/harvester/files/12511035/harvester-lhlkv.zip)
[harvester-6f5mk.zip](https://github.com/harvester/harvester/files/12511037/harvester-6f5mk.zip)
[harvester-q4vhd.zip](https://github.com/harvester/harvester/files/12511032/harvester-q4vhd.zip)
[harvester-mffzv.zip](https://github.com/harvester/harvester/files/12511030/harvester-mffzv.zip)
Others:
[part-1-supportbundle_abde46de-d3f6-4e03-bf98-a01938b43948_2023-09-04T06-49-54Z.zip](https://github.com/harvester/harvester/files/12511026/part-1-supportbundle_abde46de-d3f6-4e03-bf98-a01938b43948_2023-09-04T06-49-54Z.zip)

**Comment by Vicente-Cheng** [comment]
We found the redundant im-e could not be drained in this environment.

problematic pvc: `r1-p1-disk-0-4tzv9`, volume: `pvc-7b120d60-1577-4716-be5a-62348271025a`

This volume was migrated successfully from node `harvester-lhlkv` to `harvester-q4vhd`.
```
2023-09-03T13:37:52.660069406Z stderr F time="2023-09-03T13:37:52Z" level=info msg="Volume pvc-7b120d60-1577-4716-be5a-62348271025a migration from harvester-lhlkv to harvester-q4vhd confirmed"
```
The volume controller on the `harvester-q4vhd` updated the new engine `pvc-7b120d60-1577-4716-be5a-62348271025a-e-08220b62` to `active`, but somehow the old engine `pvc-7b120d60-1577-4716-be5a-62348271025a-e-1cd53c57` does not be cleaned up.

Due to this engine, the im-e could not be drained. We see the draining is stuck.

**Comment by irishgordo** [comment]
@Vicente-Cheng 
Also noticing this on an air-gapped upgrade, with 3 nodes:
v1.1.2 -> v1.2.0-rc6
[scc_harvester-node-0_230906_0019.zip](https://github.com/harvester/harvester/files/12529480/scc_harvester-node-0_230906_0019.zip)

**Comment by Vicente-Cheng** [comment]
The document PR: https://github.com/harvester/docs/pull/427

**Comment by guangbochen** [comment]
The workaround solution is added to https://docs.harvesterhci.io/v1.2/upgrade/v1-1-2-to-v1-2-0/#53-the-instance-manager-e-pod-could-not-be-drained, move the issue to the future milestone to tracking the upstream LH issue https://github.com/longhorn/longhorn/issues/6642.

**Comment by lanfon72** [comment]
close as resolved that unable to reproduce this case in `v1.2-head` upgrade to `v1.3.0-rc2` around 5 times.
